app/assets/javascripts/tabulatr/_events.js

```diff
@@ -1,263 +1,206 @@
     if($('.pagination[data-table="'+ $(el).attr('id') +'"]').length === 0){
       $('.pagination_trigger[data-table="'+ $(el).attr('id') +'"]').bind('inview', cbfn);
     }
   });
-
-  $('.batch-action-inputs').click(function(){
-    var a = $(this);
-    var name = a.data('do-batch-action-name');
-    var key = a.data('do-batch-action');
-    var tableId = a.data('table-id');
-    var params = {page: 1};
-    var use_ajax = !a.data('download');
+*/
+
+  $(document).on('click', '.batch-action-inputs', function() {
+    const a = $(this);
+    const name = a.data('do-batch-action-name');
+    const key = a.data('do-batch-action');
+    const table_id = a.data('table-id');
+    const table_instance = TabulatrInstances.instance.table(table_id);
+    const params = {page: 1};
+    const use_ajax = !a.data('download');
     params[name] = key;
     params.tabulatr_checked = {checked_ids: jQuery.map($('#'+ tableId +' .tabulatr-checkbox:checked'), function(el){return $(el).val();}).join(',')};
-    var confirmation = true;
-    if(params.tabulatr_checked.checked_ids === ''){
+    let confirmation = true;
+    if(params.tabulatr_checked.checked_ids === '') {
       confirmation = window.confirm(a.parents('ul').data('confirm-text'));
     }
-    if(confirmation){
-      $('.tabulatr_mark_all[data-table='+ tableId +']').prop('indeterminate', false).prop('checked', false);
-      $('#'+ tableId +' .tabulatr-wrench').addClass('disabled');
-      var table_obj;
-      for(var i = 0; i < tabulatr_tables.length; i++){
-        if(tabulatr_tables[i].id === tableId){
-          table_obj = tabulatr_tables[i];
-        }
-      }
+    if(confirmation) {
+      $('.tabulatr_mark_all[data-table='+ table_instance.id +']').prop('indeterminate', false).prop('checked', false);
+      $('#'+ table_instance.id +' .tabulatr-wrench').addClass('disabled');
       if (use_ajax)
-        table_obj.updateTable(params, true);
+        table_instance.updateTable(params, true);
       else
-        table_obj.sendRequestWithoutAjax(params, a.data('extension'));
+        table_instance.sendRequestWithoutAjax(params, a.data('extension'));
     }
   });
 
-  $('form.tabulatr-fuzzy-search').submit(function(){
-    var tableId = $(this).data('table');
-    var table_obj;
-    for(var i = 0; i < tabulatr_tables.length; i++){
-      if(tabulatr_tables[i].id === tableId){
-        table_obj = tabulatr_tables[i];
-      }
-    }
-    if(table_obj.hasInfiniteScrolling){
-      $('.pagination_trigger[data-table='+ tableId +']').unbind('inview', cbfn);
-      $('.pagination_trigger[data-table='+ tableId +']').bind('inview', cbfn);
+  $(document).on('submit', 'form.tabulatr-fuzzy-search', function() {
+    const table_id = $(this).data('table');
+    const table_instance = TabulatrInstances.instance.table(table_id);
+    if(table_instance.hasInfiniteScrolling) {
+      $('.pagination_trigger[data-table='+ table_instance.id +']').unbind('inview', cbfn);
```

Comment:
There is no need for either. It all needs to just use `.on` but with `document` to [bubble up events](https://github.com/mlt/tabulatr2/blob/webpacker/app/assets/javascripts/tabulatr/_events.js#L13) to play nice with Turbolinks.

Reply:
If you put it do document there is nowhere to bubble, since this is the root. If you attach it correctly to the corresponding elements it will bubble up each parent until `document` is reached, unless you call `.stopPropagation()`, which sometimes is what you want.

Reply:
I presume it bubbles up from the element. Anyhow.. [it is working as is](https://github.com/mlt/tabulatr2/blob/webpacker/app/assets/javascripts/tabulatr/_events.js#L36) and [passes](https://travis-ci.com/mlt/tabulatr2) all the [tests](https://github.com/mlt/tabulatr2/blob/webpacker/spec/features/tabulatrs_spec.rb#L125). Otherwise there is a need for off/on flip.